src/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -55,68 +55,17 @@ private void initialize(Collection<JexlVCMatchExp> jexlCollection) {
-
-            if ( vc != null ) {
-                // create a mapping of what we know about the variant context, its Chromosome, positions, etc.
-                infoMap.put("CHROM", vc.getChr());
-                infoMap.put("POS", vc.getStart());
-                infoMap.put("TYPE", vc.getType().toString());
-                infoMap.put("QUAL", String.valueOf(vc.getPhredScaledQual()));
-
-                // add alleles
-                infoMap.put("ALLELES", GeneralUtils.join(";", vc.getAlleles()));
-                infoMap.put("N_ALLELES", String.valueOf(vc.getNAlleles()));
-
-                // add attributes
-                addAttributesToMap(infoMap, vc.getAttributes());
-
-                // add filter fields
-                infoMap.put("FILTER", vc.isFiltered() ? "1" : "0");
-                for ( Object filterCode : vc.getFilters() ) {
-                    infoMap.put(String.valueOf(filterCode), "1");
-                }
-
-                // add genotype-specific fields
-                // TODO -- implement me when we figure out a good way to represent this
-                //    for ( Genotype g : vc.getGenotypes().values() ) {
-                //        String prefix = g.getSampleName() + ".";
-                //        addAttributesToMap(infoMap, g.getAttributes(), prefix);
-                //        infoMap.put(prefix + "GT", g.getGenotypeString());
-                //    }
-
-                // add specific genotype if one is provided
-                infoMap.put(VCFConstants.GENOTYPE_KEY, g.getGenotypeString());
-                infoMap.put("isHom", g.isHom() ? "1" : "0");
-                infoMap.put("isHomRef", g.isHomRef() ? "1" : "0");
-                infoMap.put("isHet", g.isHet() ? "1" : "0");
-                infoMap.put("isHomVar", g.isHomVar() ? "1" : "0");
-                infoMap.put("isCalled", g.isCalled()? "1" : "0");
-                infoMap.put("isNoCall", g.isNoCall()? "1" : "0");
-                infoMap.put("isMixed", g.isMixed()? "1" : "0");
-                infoMap.put("isAvailable", g.isAvailable()? "1" : "0");
-                infoMap.put("isPassFT", g.isFiltered()? "0" : "1");
-                infoMap.put(VCFConstants.GENOTYPE_FILTER_KEY, g.isFiltered()?  g.getFilters() : "PASS");
-
-                infoMap.put(VCFConstants.GENOTYPE_QUALITY_KEY, g.getGQ());
-                if ( g.hasDP() )
-                    infoMap.put(VCFConstants.DEPTH_KEY, g.getDP());
-                for ( Entry<String, Object> e : g.getExtendedAttributes().entrySet() ) {
-                    if ( e.getValue() != null && !e.getValue().equals(VCFConstants.MISSING_VALUE_v4) )
-                        infoMap.put(e.getKey(), e.getValue());
-                }
+        // QUESTION: the original code could have created a new VariantJEXLContext with a null vc.
+        // i dont think this is right, though i have no idea if vc could actually be null
+        if ( vc != null ) {
```

Comment:
I'd prefer if you kept the nesting lower by inverting the if statement and returning Collections.emptyMap() if vc==null.